pkg/sidecar/sidecar.go

```diff
@@ -183,6 +189,101 @@ func (s *Sidecar) dumpBundles(svidResponse *workloadapi.X509Context) error {
 	return nil
 }
 
+func (s *Sidecar) readJSON(fileName string) map[string]interface{} {
+	jsonPath := path.Join(s.config.CertDir, fileName)
+	file, err := os.ReadFile(jsonPath)
+	if err != nil {
+		s.config.Log.Warnf("Unable to read json file: %v", err)
+	}
+
+	certs := make(map[string]interface{})
+	err = json.Unmarshal(file, &certs)
+	if err != nil {
+		s.config.Log.Warnf("Unable to parse json: %v", err)
+	}
+
+	return certs
+}
+
+func (s *Sidecar) writeJSON(fileName string, certs map[string]interface{}) {
+	file, err := json.Marshal(certs)
+	if err != nil {
+		s.config.Log.Warnf("Unable to parse certs: %v", err)
+	}
+
+	jsonPath := path.Join(s.config.CertDir, fileName)
+	err = os.WriteFile(jsonPath, file, os.ModePerm)
+	if err != nil {
+		s.config.Log.Warnf("Unable to write JSON file: %v", err)
+	}
+}
+
+func (s *Sidecar) updateJWTBundle(jwkSet *jwtbundle.Set) {
+	s.config.Log.Info("Updating JWK bundles")
+
+	bundles := make(map[string]string)
+	for _, bundle := range jwkSet.Bundles() {
+		bytes, err := bundle.Marshal()
+		if err != nil {
+			s.config.Log.Warnf("Unable to marshal JWK bundle: %v", err)
+			continue
+		}
+		bundles[bundle.TrustDomain().Name()] = base64.StdEncoding.EncodeToString(bytes)
+	}
+
+	certs := s.readJSON(s.config.JWKFilename)
+	certs["bundles"] = bundles
```

Comment:
similar question as for the svid file, shouldn't this file only contain the bundle contents?

Reply:
Well, the origin of the PR is the https://github.com/spiffe/spiffe-helper/issues/43, In this case, the bundle could contain multiple trust domain. So I see useful the possibility to get the bundles o a json, using the trust domain as a key. Any thoughts here @faisal-memon ?

Reply:
Same JSON, just removing the top level "bundles" was the implementation I was thinking of FWIW.